File: src/Systems/Fortran/patterns/leader_followers.f90
module leader_followers_example
  implicit none
contains
  logical function run()
    character(len=8) :: workers(3)
    character(len=1) :: events(3)
    character(len=64) :: handled
    integer :: i
    workers = ['worker-1','worker-2','worker-3']; events = ['a','b','c']; handled = ''
    do i = 1, 3
      if (i > 1) handled = trim(handled)//'>'
      handled = trim(handled)//trim(workers(i))//':'//events(i)
    end do
    run = trim(handled) == 'worker-1:a>worker-2:b>worker-3:c'
  end function
end module

File: src/Systems/Swift/patterns/LeaderFollowers.swift
enum LeaderFollowersExample {
    static func run() -> Bool {
        let workers = ["worker-1", "worker-2", "worker-3"]
        let events = ["a", "b", "c"]
        let handled = events.enumerated().map { "\(workers[$0.offset % workers.count]):\($0.element)" }
        return handled.joined(separator: ">") == "worker-1:a>worker-2:b>worker-3:c"
            && workers[events.count % workers.count] == "worker-1"
    }
}
